e2e/utils/TestUtils.ts

```diff
@@ -254,6 +254,28 @@ export async function tool(tool: string, ...args: string[]): Promise<string> {
   }
 }
 
+export async function waitForRestartVM(progressBar: Locator, options = { timeout: 10_000, interval: 200 }): Promise<void> {
+  const startTime = new Date().valueOf();
+  const endTime = startTime + options.timeout;
+
+  await progressBar.waitFor({ state: 'visible', timeout: 10_000 });
+  console.log(`Waiting for RD to restart the VM...`);
+  while (true) {
+    const caption: string = await progressBar.textContent() ?? '';
+
+    if (caption.startsWith('Starting virtual machine')) {
+      console.log(`Restart detected.`);
+      break;
+    }
+    const nowTime = new Date().valueOf();
+
+    if (nowTime > endTime) {
+      break;
```

Comment:
IMO we should throw an error here saying something like "timed out waiting for VM startup to start".

Reply:
good idea